### PR TITLE
[14.0] product_analytic_sale

### PR DESCRIPTION
**Pull request by alexis-via**
The readme gives an example of a scenario where this module product_analytic_sale is required.

### CODE REVIEW

**Review by clementmbr — APPROVED** [review]
LGTM

**Review by rvalyi — APPROVED** [review]

**Comment by clementmbr** [comment]
@alexis-via I just found a bug in case the Sale Order Line as no product_id (type note). I added a commit to fix it.
cc @rvalyi

File: product_analytic_sale/models/sale_order_line.py [hunk 1]
```diff
@@ -10,9 +10,12 @@ class SaleOrderLine(models.Model):
 
     def _prepare_invoice_line(self, **optional_values):
         vals = super()._prepare_invoice_line(**optional_values)
-        ana_account = self.product_id.product_tmpl_id._get_product_analytic_accounts()[
-            "income"
-        ]
-        if ana_account:
-            vals["analytic_account_id"] = ana_account.id
+        if self.product_id:
+            ana_account = (
```
Comment (rousseldenis) [inline product_analytic_sale/models/sale_order_line.py:14]: Nitpicking, but I'd prefer analytic_account than the short (less significant) term

**Comment by rousseldenis** [comment]
It should be great to add a little test.

**Review by rafaelbn — APPROVED** [review]
This works 👍 

As @rousseldenis , it could be great a little test https://github.com/OCA/account-analytic/pull/462#issuecomment-1154109242

My curiosity / question is , why not improve product_analytic with this so we don't need glue modules? I complicated so we need another module

**Comment by rousseldenis** [comment]
> This works +1
> 
> As @rousseldenis , it could be great a little test [#462 (comment)](https://github.com/OCA/account-analytic/pull/462#issuecomment-1154109242)
> 
> My curiosity / question is , why not improve product_analytic with this so we don't need glue modules? I complicated so we need another module

@rafaelbn This to avoid a strong dependency on sale in product_analytic

**Pull request merged**